corehq/apps/notifications/models.py

```diff
@@ -34,7 +36,8 @@ class Meta(object):
         ordering = ["-activated"]
 
     @classmethod
-    def get_by_user(cls, django_user, couch_user, limit=10):
+    def get_by_user(cls, django_user, couch_user, limit=10, plan='Community',
+                    has_cs_groups=False, is_USH_or_Solutions=False):
```

Comment:
Additionally, I wonder if there is a more appropriate name than `has_cs_groups`. If `is_USH_or_Solutions` is equivalent to `should_exclude_features`, `has_cs_groups` seems to function similarly, but specific to advanced plans. Maybe you could pull that logic out into the helper that is responsible for determining if feature discoverability notifications should be excluded in the first place, so then you'd simplify the interface for this method slightly.